src/wallet/components/WalletModal.tsx

```diff
@@ -0,0 +1,296 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { useConnect, useConnectors } from 'wagmi';
+import { coinbaseWallet } from 'wagmi/connectors';
+import { closeSvg } from '../../internal/svg/closeSvg';
+import { coinbaseWalletSvg } from '../../internal/svg/coinbaseWalletSvg';
+import { defaultAvatarSVG } from '../../internal/svg/defaultAvatarSVG';
+import { walletConnectSvg } from '../../internal/svg/walletConnectSvg';
+import {
+  background,
+  border,
+  cn,
+  color,
+  line,
+  pressable,
+  text,
+} from '../../styles/theme';
+import { useOnchainKit } from '../../useOnchainKit';
+
+type WalletModalProps = {
+  isOpen: boolean;
+  onClose: () => void;
+  className?: string;
+  onError?: (error: Error) => void;
+};
+
+export function WalletModal({
+  isOpen,
+  onClose,
+  className,
+  onError,
+}: WalletModalProps) {
+  const modalRef = useRef<HTMLDivElement>(null);
+  const [shouldRender, setShouldRender] = useState(isOpen);
+  const { connect } = useConnect();
+  const connectors = useConnectors();
+  const { config } = useOnchainKit();
+
+  useEffect(() => {
+    if (isOpen) {
+      setShouldRender(true);
+    }
+  }, [isOpen]);
+
+  const handleAnimationEnd = () => {
+    if (!isOpen) {
+      setShouldRender(false);
+    }
+  };
+
+  const appLogo = config?.appearance?.logo;
+  const appName = config?.appearance?.name;
+  const privacyPolicyUrl = config?.wallet?.privacyUrl ?? undefined;
+  const termsOfServiceUrl = config?.wallet?.termsUrl ?? undefined;
+
+  const handleCoinbaseWalletConnection = useCallback(() => {
+    try {
+      const cbConnector = coinbaseWallet({
+        preference: 'all',
+      });
+
+      connect({ connector: cbConnector });
```

Comment:
Could you update this component to accept `onConnect` as a prop, then pass it to the `onSuccess` callback here? That would be a lot cleaner than the solution I initially implemented in `ConnectWallet` where we have to track `hasClickedConnect`